### PR TITLE
Set up basic law-checking.

### PR DESCRIPTION
**Pull request by non**
This commit does a lot of things:

 1. Sets up Discipline laws for Invariant through Applicative
 2. Uses ScalaTest to verify the laws for some instances
 3. Fixes a weird issue with Bijection's compose method
 4. Introduces ArbitraryK[F[_]] for generating Arbitrary[F[A]]s.
 5. Starts integrating algebra's std instances into cats.std.

There are still a lot of issues that need fixing:

 1. We should actually remove Bijection.
 2. Ideally we'd move from ScalaTest to specs2 or similar.
 3. Full algebra.std integration once its instances are more complete.
 4. Determine how many concrete A, B, and C types we want to test for.
 5. Add all the laws!
 6. Ensure that we're exporting the laws in a useful way.

Whew!

### CODE REVIEW

**Comment by non** [comment]
(I am happy to leave this branch open while adding more laws, or we could merge it if we want to get an example of doing law-checking into master sooner. Either option seems OK with me.)

File: std/build.sbt [hunk 1]
```diff
@@ -1 +1,5 @@
 name := "std"
+
+libraryDependencies in ThisBuild ++=
```
Comment (travisbrown) [inline std/build.sbt]: Doesn't the `in ThisBuild` here add this dependency to the entire build? If I run `sbt "project core" console` I can import `algebra.std._`.

Is there a reason to split the build config up like this (or even to have both a root-level `build.sbt` and `project/build.scala`)?

Comment (non) [inline std/build.sbt]: Maybe not?

At the time it seemed like a good idea but it's becoming clear that I don't know what I'm doing.

**Comment by non** [comment]
(By the way, I expect to get an algebra milestone on Sonatype next week. So these URLs should only be around for a short time.)

**Comment by travisbrown** [comment]
:+1: from me, but about the SBT change: I think the sbt-release parts are a direct translation from the old format, but I didn't try them out, so they might need a little tinkering.

**Comment by non** [comment]
@travisbrown yeah I'm fairly sure I got that stuff right but we'll never know until we try to do a release.

**Comment by stew** [comment]
:+1:

**Comment by non** [comment]
Alright with +2 I'll go ahead and merge.

**Pull request merged**